catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersion.java

```diff
@@ -196,6 +233,20 @@ public MetacardVersion(Metacard sourceMetacard, Action action, Subject subject)
         this.setVersionOfId(sourceMetacard.getId());
         this.setVersionTags(sourceMetacard.getTags());
 
+        Optional<MetacardType> type = types.stream()
+                .filter(mt -> sourceMetacard.getMetacardType()
+                        .getName()
+                        .equals(mt.getName()))
+                .findFirst();
+
+        this.setVersionType(type.get()
```

Comment:
Unsafe `get()` call. Also, is there a reason why the binary is only being set in one of the cases?